modules/perception/lidar/lib/detection/lidar_point_pillars/point_pillars_detection.cc

```diff
@@ -71,42 +76,131 @@ bool PointPillarsDetection::Detect(const DetectionOptions& options,
   }
 
   // transform point cloud into an array
-  float* points_array = new float[original_cloud_->size() * 4];
-  PclToArray(original_cloud_, points_array, kNormalizingFactor);
+  float* points_array;
+  int num_points = original_cloud_->size();
+  int num_point_indexes;
+  std::vector<int> point_indexes;
+  if (kFuseFrames && kNumFuseFrames > 1) {
+    // before fusing
+    while (!prev_point_clouds_.empty() &&
+        frame->timestamp - prev_point_clouds_.front().second > kTimeInterval) {
+      prev_point_clouds_.pop_front();
+    }
+
+    // fusing
+    for (auto& pc_timestamp : prev_point_clouds_) {
+      num_points += pc_timestamp.first->size();
+    }
+    num_point_indexes = num_points;
+    point_indexes = GenerateIndexes(0, num_point_indexes, kShufflePoints);
+    num_points = std::min(num_points, kMaxNumPoints);
+    points_array = new float[num_points * kNumPointFeature];
+    memset(points_array, 0, num_points * kNumPointFeature * sizeof(float));
```

Comment:
In c++, you can just write `points_array = new float[N]()` to allocate an initialized array.

Reply:
@jinghaomiao `points_array` is required by a cuda function in 'point_pillars.cc' which doesn't support c++ std type.

Reply:
Good to know. Please double check if all memory is safely released in all cases to avoid memory leak.